app/models/mixins/miq_request_mixin.rb

```diff
@@ -187,7 +187,7 @@ def dialog_zone
 
   def mark_execution_servers
     options[:executed_on_servers] ||= []
-    options[:executed_on_servers] << MiqServer.my_server.id
+    (options[:executed_on_servers] -= [MiqServer.my_server.id]) << MiqServer.my_server.id
```

Comment:
Can you add an comment explaining that we're not only removing all copies of the server id but also ensuring it's the last entry?

Reply:
ok, done